sound/soc/sof/ipc.c

```diff
@@ -491,12 +491,49 @@ static void ipc_xrun(struct snd_sof_dev *sdev, u32 msg_id)
 #endif
 }
 
+static int kwd_notification_handler(struct snd_sof_dev *sdev, u32 comp_id)
+{
+	int ret = -ENODEV;
+	int direction = SNDRV_PCM_STREAM_CAPTURE;
+	int cmd = SNDRV_PCM_TRIGGER_START;
+	struct snd_sof_pcm *spcm;
+	struct snd_pcm_substream *substream;
+
+	/* Firmware detected a key phrase and is about to start draining.
+	 * Its timme to resume the KWD stream now.
+	 *
+	 * Find correct stream first - KWD stream has suspend_ignored
+	 * attribute assigned.
+	 */
+	list_for_each_entry(spcm, &sdev->pcm_list, list) {
+		if (spcm->stream[direction].suspend_ignored) {
```

Comment:
@mrajwa If there ever were more than one detection algorithms running in the FW, you'd start them all. Sorry this wouldnt scale.

Reply:
Yep, we need to pass the relevant stream data in sof_ipc_comp_event.

Reply:
Hmm but if there is more than one detection algorithm you also want to resume its stream right?